src/hotspot/share/classfile/loaderConstraints.cpp

```diff
@@ -447,6 +443,18 @@ InstanceKlass* LoaderConstraintTable::find_constrained_klass(Symbol* name,
   return nullptr;
 }
 
+// Removes a class that was added to the table then class loading subsequently failed for this class.
+void LoaderConstraintTable::remove_failed_loaded_klass(InstanceKlass* klass,
+                                                       ClassLoaderData* loader) {
+
+  Symbol* name = klass->name();
+  LoaderConstraint *p = find_loader_constraint(name, loader);
+  if (p != nullptr && p->klass() != nullptr && p->klass() == klass) {
+    log_info(class, loader, constraints)("removing klass %s failed to load", name->as_C_string());
```

Comment:
Suggestion, either:
```
"removing klass %s: failed to load"
```
or
```
"removing klass %s as it failed to load"
```

Reply:
Actually "failed to load" seems rather vague here, the actual condition is much more specific isn't it?

Reply:
The actual condition is in the pending exception and only in the case of OOM will we actually remove the entry.  I know this but the code is more general.

Reply:
I'll update the comments.